src/position.cpp

```diff
@@ -839,6 +839,9 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
   // Set capture piece
   st->capturedPiece = captured;
 
+  // Draw at first repetition
+  st->draw = true;
```

Comment:
Well, if I find a way to access the root position from Position::is_draw() I will create a fishtest test to compare the two approaches. I already did that for syzygy's idea and it turned out to be not good :).